src/main/java/be/acara/events/service/EventService.java

```diff
@@ -54,7 +56,25 @@ private Event getEvent(long id) {
     }
 
     public EventDto addEvent(EventDto eventDto) {
+        if (eventDto.getId() != null) {
+            throw new IdAlreadyExistsException("A new entity cannot already contain an id");
+        }
         Event event = mapper.map(eventDto);
         return mapper.map(repository.saveAndFlush(event));
     }
+
+    public EventDto editEvent(long id, EventDto eventDto) {
+        Event eventToEdit = getEvent(id);
+        if(eventDto.getId().equals(eventToEdit.getId())){
+            Event event = mapper.map(eventDto);
+            return mapper.map(repository.saveAndFlush(event));
+        }
+        else {
+            throw new BadRequestException(
+                    "Bad request on editing member",
+                    String.format(
+                            "Id of member to edit does not match given id. Member id = %d, and given id = %d", eventDto.getId(), id)
+            );
+        }
```

Comment:
Flip else-if to eliminate the else. Throw if ids aren't equal, otherwise continue as usual.
(this code isn't wrong, it's a more a clean code principle)